cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/ProgressiveSolver.scala

```diff
@@ -105,7 +105,9 @@ class ProgressiveSolver(val dataManager: ActorRef,
             "end" -> JsNumber(boundary.getEnd().getMillis())
         ))
         // for query with slicing request, add current timeInterval information in its query results.
-        val infoValue = queryGroup.postTransform.transform(JsArray(returnedResult))
+        val avgHandledResult = QueryPlanner.handleAvg(returnedResult)
```

Comment:
For every query, do we need to go through this function?

Reply:
Solved by inside `handleAvg` function check whether we need go through records or not.